orderer/common/multichannel/registrar.go

```diff
@@ -1025,8 +1063,10 @@ func (r *Registrar) removeSystemChannel() error {
 	}
 
 	// remove system channel resources
-	err := r.ledgerFactory.Remove(systemChannelID)
+	r.pendingRemoval[systemChannelID] = true
+	err := r.ledgerFactory.Remove(systemChannelID, r.removeChannelFromPendingRemovals)
 	if err != nil {
+		r.pendingRemoval[systemChannelID] = false
```

Comment:
Remove the system channel could be done synchronously, that would save us a ton of trouble.